test/general/backend_inclusion.pass.cpp

```diff
@@ -0,0 +1,151 @@
+#   endif
+#endif
+
+#if defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_TBB_BACKEND
+#   ifdef _ONEDPL_PARALLEL_BACKEND_TBB_H
+#       error The parallel TBB backend is used while it should not (ONEDPL_USE_TBB_BACKEND==0)
+#   endif
+#endif
+
+#if defined(ONEDPL_USE_DPCPP_BACKEND) && !ONEDPL_USE_DPCPP_BACKEND
+#   ifdef _ONEDPL_parallel_backend_sycl_H
+#       error The parallel DPC++ backend is used while it should not (ONEDPL_USE_DPCPP_BACKEND==0)
+#   endif
+#   ifdef _ONEDPL_parallel_backend_sycl_fpga_H
+#       error The DPC++ backend for the FPGA is used while it should not (ONEDPL_USE_DPCPP_BACKEND==0)
+#   endif
+#endif
+
+// Verify that DPC++ backend for the FPGA is not not accessed if ONEDPL_FPGA_DEVICE is undefined or set to zero.
+#if !ONEDPL_FPGA_DEVICE
+#   ifdef _ONEDPL_parallel_backend_sycl_fpga_H
+#       error The DPC++ backend for the FPGA is used while it should not (ONEDPL_FPGA_DEVICE==0)
+#   endif
+#endif
+
+// Verify there is only one backend selected and the selection in the table below.
+//  ___________________________________________________________
+// |       \      |                     |           |          |
+// | OpenMP \ TBB |          0          | Undefined |    1     |
+// |_________\____|_____________________|___________|__________|
+// |              |                     |           |          |
+// |      0       |       Serial        |    TBB    |   TBB    |
+// |______________|_____________________|___________|__________|
+// |              | OpenMP if available |           |          |
+// |  Undefined   |  Otherwise, serial  |    TBB    |   TBB    |
+// |______________|____________________ |___________|__________|
+// |              |                     |           |          |
+// |       1      |       OpenMP        |  OpenMP   |   TBB    |
+// |______________|_____________________|___________|__________|
+
+// Make sure that the TBB backend is selected if ONEDPL_USE_TBB_BACKEND set to 1 and ONEDPL_USE_OPENMP_BACKEND set to any value
+#if defined(ONEDPL_USE_TBB_BACKEND) && ONEDPL_USE_TBB_BACKEND
+#   if !defined(_ONEDPL_PARALLEL_BACKEND_TBB_H)
+#       error The parallel TBB backend is not used, although it should (ONEDPL_USE_TBB_BACKEND==1)
+#   endif
+#   if defined(_ONEDPL_PARALLEL_BACKEND_OMP_H)
+#       error The parallel OpenMP backend is used, although it should not, because TBB backend is used
+#   endif
+#   if defined(_ONEDPL_PARALLEL_BACKEND_SERIAL_H)
+#       error The serial backend is used, although it should not, because TBB backend is used
+#   endif
+#endif
+
+
+// Make sure that the OpenMP backend is selected if ONEDPL_USE_OPENMP_BACKEND is set to 1
+// and ONEDPL_USE_TBB_BACKEND is undefined or set to 0
+#if defined(ONEDPL_USE_OPENMP_BACKEND) && ONEDPL_USE_OPENMP_BACKEND && !ONEDPL_USE_TBB_BACKEND
+#   if defined(_ONEDPL_PARALLEL_BACKEND_TBB_H)
+#       error The parallel TBB backend is used, although it should not, because OpenMP backend is used

```

Comment:
all such phases should be rewritten. I don't have the immediate suggestion. Need to think about that more. But currently it's doesn't say the reason it just says that "_one_ **is used**, but should not because the _other_ **is used**", which is not helpful